devito/symbolics/extended_sympy.py

```diff
@@ -394,13 +395,29 @@ def __new__(cls, base, dtype=None, stars=None, reinterpret=False, **kwargs):
             # E.g. void
             pass
 
+        dtype, stars = cls._process_dtype(dtype, stars)
+
         obj = super().__new__(cls, base)
         obj._stars = stars or ''
         obj._dtype = dtype
         obj._reinterpret = reinterpret
 
         return obj
 
+    @classmethod
+    def _process_dtype(cls, dtype, stars):
+        if not isinstance(dtype, str) or stars is not None:
+            return dtype, stars
+
+        # String dtype, e.g. "float", "int*", "foo**"
```

Comment:
so `stars` cannot be None, hence maybe the first example "float" is not possible?